.github/workflows/publish.yml

```diff
@@ -0,0 +1,21 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - "main"
```

Comment:
I don't think we should be publishing a new version on every push to `main`.  this should be controlled by a tag.

also, is there a reason why you went with `skaffold` for building and pushing?  we've been using `goreleaser` within the rode org for this, since it can produce both docker images and binaries.  I'd prefer if we switched to this unless there is a good reason to use `skaffold` for this.

Reply:
Gotcha, Skaffold seemed to be used over on the [collector-sonarqube](https://github.com/rode/collector-sonarqube/blob/main/.github/workflows/publish.yml), so was just going based off of that. I can go in and change it to use goreleaser

Reply:
Yeah if we're using some sort of template on our collectors, we should change it to use goreleaser.  While we may run these in containers on Kubernetes, we should also be building and publishing binaries for those who are not.